File: sdks/c++/basys.sdk.cc/src/submodel/submodel/map/dataspecification/DataSpecificationContent.cpp
/*
 * DataSpecificationContent.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/map/dataspecification/DataSpecificationContent.h>

namespace basyx {
namespace submodel {

DataSpecificationContent::DataSpecificationContent(basyx::object obj)
  : vab::ElementMap(obj)
{}

DataSpecificationContent::DataSpecificationContent(const IDataSpecificationContent & other)
  : vab::ElementMap()
{
}

}
}

File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/map/dataspecification/DataSpecificationContent.java
package org.eclipse.basyx.submodel.metamodel.map.dataspecification;

import java.util.Map;

import org.eclipse.basyx.submodel.metamodel.api.dataspecification.IDataSpecificationContent;
import org.eclipse.basyx.vab.model.VABModelMap;

public class DataSpecificationContent extends VABModelMap<Object> implements IDataSpecificationContent {
	/**
	 * Creates a DataSpecificationContent object from a map
	 * 
	 * @param obj
	 *            a DataSpecificationContent object as raw map
	 * @return a DataSpecificationContent object, that behaves like a facade for the given map
	 */
	public static DataSpecificationContent createAsFacade(Map<String, Object> map) {
		if (map == null) {
			return null;
		}

		DataSpecificationContent ret = new DataSpecificationContent();
		ret.setMap(map);
		return ret;
	}
}
